docs/USAGE-GUIDE.md

```diff
@@ -1704,116 +1704,6 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
 **s2n_config_add_ticket_crypto_key** adds session ticket key on the server side. It would be ideal to add new keys after every (encrypt_decrypt_key_lifetime_in_nanos/2) nanos because
 this will allow for gradual and linear transition of a key from encrypt-decrypt state to decrypt-only state.
 
-### Asynchronous private key operations related calls
-
-When s2n-tls is used in non-blocking mode, this set of functions allows user
-to move execution of CPU-heavy private key operations out of the main
-event loop, preventing **s2n_negotiate** blocking the loop for a few
-milliseconds each time the private key operation needs to be performed.
-
-To enable asynchronous private key operations user needs to provide a
-callback function **s2n_async_pkey_fn** to
-**s2n_config_set_async_pkey_callback** call. This function will be
-executed during **s2n_negotiate** call every time an operation on private
-key needs to be performed. The argument **op** represents the operation
-to perform. From the callback the user can spawn the thread to perform
-**op** through **s2n_async_pkey_op_perform** call and immediately return
-**S2N_SUCCESS** from the function without waiting for thread to complete.
-The **s2n_negotiate** will return **S2N_FAILURE** with **S2N_ERR_T_BLOCKED**
-error type and **s2n_blocked_status** **S2N_BLOCKED_ON_APPLICATION_INPUT**,
-and will keep giving the same error until the **op** is performed and
-applied to the connection through **s2n_async_pkey_op_apply** call.
-
-Note, it is not safe to call multiple functions on the same **conn** or
-**op** objects from 2 different threads at the same time. Doing so will
-produce undefined behavior. However it is safe to have a call to
-function involving only **conn** at the same time with a call to
-function involving only **op**, as those 2 objects are not coupled with
-each other. It is also safe to free **conn** or **op** at any moment with
-respective function calls, with the only exception that **conn** cannot
-be freed inside the **s2n_async_pkey_fn** callback.
-
-```c
-typedef int (*s2n_async_pkey_fn)(struct s2n_connection *conn, struct s2n_async_pkey_op *op);
```

Comment:
So the list of APIs are getting deleted?

Reply:
I think its contents belonged in doxygen comments. I struggled to integrate the list into a discussion of how to use the APIs, since they don't really fall neatly into the three sections I ended up with. Did you think the list was helpful?

Reply:
Not sure, I just know that's how we usually write our docs.